### DISCUSSION TITLE
Is there a way to do this with a soil moisture sensor?

### DISCUSSION THREAD

**Discussion by RandomPotato893** [RavenSystem/esp-homekit-devices#1646] [General]
Hello, I was just wondering what the json would be for a capacitive soil moisture sensor, or if it is even possible?
Thanks,
Sam

**Comment by cgrylmz** [comment]
yea, i would like to know that too!

**Comment by paulzhdanov** [comment]
@RandomPotato893 I failed to get an [answer](https://github.com/RavenSystem/esp-homekit-devices/discussions/1537), but I just bought Sonoff MS01 (capacitive), it works perfect with native sonoff firmware, so I will try to analyze the log to understand how to connect it with HAA. In worst case scenario, I will use it with Raspberry HomeBridge with eWelink plugin, it works ok.
BTW, there is no moisture type in HK, humidity only.

**Comment by zerodago** [comment]
I am also very interested in resolving this connection with ms01. thks.

**Comment by zerodago** [comment]
Would anyone have the original firmware for the th16? ths
